gen_schema.py

```diff
@@ -0,0 +1,41 @@
+from typing import Annotated, Any, Sequence
+
+from pydantic import Field, schema, schema_json_of
+from pydantic.fields import ModelField
+from pydantic.schema import SkipField
+
+from kpops.components.base_components.kafka_connect import (
+    KafkaSinkConnector,
+    KafkaSourceConnector,
+)
+from kpops.components.streams_bootstrap.producer.producer_app import ProducerApp
+from kpops.components.streams_bootstrap.streams.streams_app import StreamsApp
+
+original_field_schema = schema.field_schema
+
+
+# adapted from https://github.com/tiangolo/fastapi/issues/1378#issuecomment-764966955
+def field_schema(field: ModelField, **kwargs: Any) -> Any:
+    if field.field_info.extra.get("hidden_from_schema"):
+        raise SkipField(f"{field.name} field is being hidden")
+    else:
+        return original_field_schema(field, **kwargs)
+
+
+schema.field_schema = field_schema
+
+PipelineComponent = StreamsApp | ProducerApp | KafkaSourceConnector | KafkaSinkConnector
```

Comment:
Should we also add `KubernetesApp`?

Reply:
good point, should we also add `KafkaApp` then?

Reply:
Yes go for it 🚀

Reply:
I've added both of them. Mypy is unfortunately complaining about the use of non-overlapping `Literal` for the `schema_type`, so I had to manually silence these errors

```
Incompatible types in assignment (expression has type "Literal['streams-app']",
base class "KafkaApp" defined the type as "Literal['kafka-app']")  [assignment]
```